File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* solution(const char* s) {

    int* answer = (int*)malloc(sizeof(int)*10000);
    int a[26];
    int i=0;
    for(i=0;i<26;i++){a[i]=-1;}
    i=0;
    while(s[i]){
        if(a[s[i]-97] < 0){
            answer[i] = -1;
        }
        else{
            answer[i] = i - a[s[i]-97];
        }
        a[s[i]-97] = i;
        i++;
    }

    return answer;
}

File: 프로그래머스/1/142086.가장 가까운 같은 글자-Python/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    a = [-1 for i in range(26)]
    for i in range(len(s)):
        n = ord(s[i]) - 97

        if a[n] < 0:
            answer.append(-1)
            a[n] = i
        else:
            answer.append(i - a[n])
            a[n] = i
    
    return answer